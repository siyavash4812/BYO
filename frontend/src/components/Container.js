import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Flex = styled.div`
  display: flex;
  justify-content: ${({ justifyContent }) => justifyContent};
  flex-wrap: wrap;
  padding: 20px;
  flex-direction: ${({ direction }) => direction};
`;

const Container = ({ children, ...props }) => {
  return <Flex {...props}>{children}</Flex>;
};

Container.propTypes = {
  justifyContent: PropTypes.oneOf([
    "center",
    "flex-end",
    "flex-start",
    "space-around",
    "space-between",
  ]),
  direction: PropTypes.oneOf([
    "row",
    "column",
    "row-reverse",
    "column-reverse",
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
Container.defaultProps = {
  justifyContent: "center",
  direction: "column",
};
export default Container;
