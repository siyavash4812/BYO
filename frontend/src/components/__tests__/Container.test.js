import renderer from "react-test-renderer";
import Container from "../Container";

it("renders flex container", () => {
  const tree = renderer
    .create(
      <Container>
        <h1>Hello</h1>
      </Container>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("display", "flex");
});
