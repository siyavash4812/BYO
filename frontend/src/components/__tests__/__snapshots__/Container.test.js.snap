// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders flex container 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
  direction="column"
>
  <h1>
    Hello
  </h1>
</div>
`;
